HomeWorkApp/src/main/java/org/example/zoo/Main.java

```diff
@@ -0,0 +1,114 @@
+package org.example.zoo;
+
+// Абстрактный класс "Животное"
+abstract class Animal {
+    protected String name; // Имя животного
+    protected int maxRunDistance; // Максимальная дистанция для бега
+    protected int maxSwimDistance; // Максимальная дистанция для плавания
+
+    // Конструктор
+    public Animal(String name, int maxRunDistance, int maxSwimDistance) {
+        this.name = name;
+        this.maxRunDistance = maxRunDistance;
+        this.maxSwimDistance = maxSwimDistance;
+    }
+
+    // Абстрактные методы для бега и плавания
+    public abstract void run(int distance);
+    public abstract void swim(int distance);
+}
+
+// Класс "Собака", наследуется от "Животное"
+class Dog extends Animal {
+    private static int dogCount = 0; // Счетчик собак
+
+    // Конструктор
+    public Dog(String name) {
+        super(name, 500, 10); // У собаки: бег 500 м., плавание 10 м.
+        dogCount++;
+    }
+
+    @Override
+    public void run(int distance) {
+        if (distance <= maxRunDistance) {
+            System.out.println(name + " пробежал " + distance + " м.");
+        } else {
+            System.out.println(name + " не смог пробежать " + distance + " м.");
+        }
+    }
+
+    @Override
+    public void swim(int distance) {
+        if (distance <= maxSwimDistance) {
+            System.out.println(name + " проплыл " + distance + " м.");
+        } else {
+            System.out.println(name + " не смог проплыть " + distance + " м.");
+        }
+    }
+
+    // Метод для получения количества собак
+    public static int getDogCount() {
+        return dogCount;
+    }
+}
+
+// Класс "Кот", наследуется от "Животное"
+class Cat extends Animal {
+    private static int catCount = 0; // Счетчик котов
+
+    // Конструктор
+    public Cat(String name) {
+        super(name, 200, 0); // У кота: бег 200 м., плавание 0 м.
+        catCount++;
+    }
+
+    @Override
+    public void run(int distance) {
+        if (distance <= maxRunDistance) {
+            System.out.println(name + " пробежал " + distance + " м.");
+        } else {
+            System.out.println(name + " не смог пробежать " + distance + " м.");
+        }
+    }
+
+    @Override
+    public void swim(int distance) {
+        System.out.println(name + " не умеет плавать.");
+    }
+
+    // Метод для получения количества котов
+    public static int getCatCount() {
+        return catCount;
+    }
+}
+
+// Основной класс программы
+public class Main {
+    private static int animalCount = 0; // Счетчик всех животных
+
+    public static void main(String[] args) {
+        // Создаем животных
+        Dog dog1 = new Dog("Корги");
```

Comment:
Не самое подходящее имя для переменной.
Имя должен определять суть данных, которые будут связаны с этой переменной. Иначе, будет сложно анализировать код и понимать его.
```suggestion
        Dog korgi = new Dog("Корги");
```